### PR TITLE
test: add suffix to filename for compare with snapshot

### PR DESCRIPTION
**Pull request by AlexeyCL**
## Brief Summary of Changes
-

## Reviewer, Please Note:
-

## Testing (what was tested)
-

### CODE REVIEW

File: src/commons/BrowserUtils.ts [hunk 1]
```diff
@@ -32,6 +32,34 @@ export interface IComparisonPath {
...
 }
 
+const sanitizeString = (str = '') =>
+  str
+    .trim()
+    .replace(/[^a-z0-9]/gi, '_')
+    .toLowerCase();
+
+//this is working because we are setting those values in a beforeTest hook in wdio config
+const getFilename = (filenameSuffix) => {
```
Comment (refael-m) [inline src/commons/BrowserUtils.ts:42]: 1. isn't it better to move it to TestUtils.ts?
2. it's not typescript

Comment (AlexeyCL) [reply src/commons/BrowserUtils.ts]: moved and added types

**Review by refael-m — APPROVED** [review]

File: src/commons/BrowserUtils.ts [hunk 2]
```diff
@@ -992,15 +994,20 @@ export namespace BrowserUtils {
...
    */
   export async function compareWithSnapshot(
-    filename: string,
+    filenameSuffix: string,
     selector: string,
     options?: WdioCheckElementMethodOptions
   ): Promise<void> {
+    const filename = getFilename(filenameSuffix);
+
+    if (!filename) {
```
Comment (FelixZilber) [inline src/commons/BrowserUtils.ts:1008]: move it to getFilename method

File: src/commons/TestUtils.ts [hunk 3]
```diff
@@ -69,4 +69,40 @@ export namespace TestUtils {
...
+   * @param filenameSuffix
+   * example of wfio.config
+   * beforeTest: function (test, context) {
+   *     global.currentTest = test;
+   *     global.currentContext = context;
+   *   },
+   */
+  export const getFilename = (filenameSuffix: string): string => {
+    const testcaseNameSanitized: string = sanitizeString(global?.currentTest?.title);
+    const filenameSanitized: string = sanitizeString(path.basename(global?.currentTest?.file || ''));
```
Comment (FelixZilber) [inline src/commons/TestUtils.ts:90]: the || '' makes if in line 92 always false

File: src/commons/BrowserUtils.ts [hunk 4]
```diff
@@ -992,15 +994,20 @@ export namespace BrowserUtils {
...
   /**
    * Comparing image by providing selector of element
-   * @param filename filename of image
+   * @param filenameSuffix suffix for filename
    * @param selector selector for an element
    * @param options https://github.com/wswebcreation/wdio-image-comparison-service/blob/main/docs/OPTIONS.md#plugin-options
    */
   export async function compareWithSnapshot(
-    filename: string,
+    filenameSuffix: string,
```
Comment (FelixZilber) [inline src/commons/BrowserUtils.ts:1002]: this should be optional

File: src/commons/TestUtils.ts [hunk 5]
```diff
@@ -69,4 +69,40 @@ export namespace TestUtils {
...
+  /**
+   * Get file name from
+   * @param filenameSuffix
+   * example of wfio.config
+   * beforeTest: function (test, context) {
+   *     global.currentTest = test;
+   *     global.currentContext = context;
+   *   },
+   */
+  export const getFilename = (filenameSuffix: string): string => {
```
Comment (FelixZilber) [inline src/commons/TestUtils.ts:88]: better to sanitize filenameSuffix as well

File: src/commons/TestUtils.ts [hunk 6]
```diff
@@ -69,4 +69,40 @@ export namespace TestUtils {
...
+    const testcaseNameSanitized: string = sanitizeString(global?.currentTest?.title);
+    const filenameSanitized: string = sanitizeString(path.basename(global?.currentTest?.file || ''));
+
+    if (!testcaseNameSanitized || !filenameSanitized) {
+      console.warn(
+        'cannot read current test title or filename, make sure you set the "currentTest" as a global variable'
+      );
+
+      return filenameSuffix || '';
```
Comment (FelixZilber) [inline src/commons/TestUtils.ts:97]: empty string can not be returned. will cause '.png' filename

**Pull request closed without merging**